### PR TITLE
Add test for `ignite generate model` and update Generator Template docs

### PR DESCRIPTION
**Pull request by joshuayoes**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR

Add tests for model generation and update docs

### CODE REVIEW

File: test/vanilla/ignite-generate.test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        /**
+         * The RootStore instance.
+         */
+        export interface RootStore extends Instance<typeof RootStoreModel> {}
+        /**
+         * The data of a RootStore.
+         */
+        export interface RootStoreSnapshot extends SnapshotOut<typeof RootStoreModel> {}
+        "
+      `)
```
Comment (jamonholmgren) [inline test/vanilla/ignite-generate.test.ts:161]: I think for this if we are going to get so specific we should just do a regular snapshot instead.

I'd probably just check for `export const RootStoreModel` and call it good, unless there's a specific reason we need to test all the other details.

**Review by jamonholmgren — APPROVED** [review]
Looks good, concerned a bit about hardcoding the generated template as it'll make maintenance harder.

File: test/vanilla/ignite-generate.test.ts (hunk 1 above)
Comment (joshuayoes) [reply test/vanilla/ignite-generate.test.ts:161]: So the nice thing about this, is that Jest automatically creates this snapshot: https://jestjs.io/docs/snapshot-testing#inline-snapshots

It's the same as `toMatchSnapshot`, but the snapshot is colocated

Comment (joshuayoes) [reply test/vanilla/ignite-generate.test.ts:161]: And the snapshot can be updated by pressing `u` in the test runner

Comment (joshuayoes) [reply test/vanilla/ignite-generate.test.ts:161]: Main thing I like about snapshots or things like this is that we get more detailed error messages when a test throws

Comment (joshuayoes) [reply test/vanilla/ignite-generate.test.ts:161]: Here is a video of what I'm talking about

https://user-images.githubusercontent.com/37849890/190252650-8603ffda-e0ef-4d8c-96aa-0fca7d599b27.mov

**Pull request merged**